### PR TITLE
Added tags display to kotest tool window

### PR DESCRIPTION
**Pull request by sksamuel**
Adds new window which scans for tags defined in files named KotestTags.kt

image

Next step will be to execute tests with those tags when you double click. We may need our (in the works) gradle plugin for that though.

### CODE REVIEW

File: src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      when (it) {
+         is KtClassOrObject -> if (it.getAllSuperClasses().contains(TagSuperClass)) it.name else null
+         is KtProperty -> it.name
+         else -> null
+      }
+   }
+}
+
+fun scanTags(project: Project) {
+   DumbService.getInstance(project).runWhenSmart {
```
Comment (Kantis) [inline src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt:49]: What does dumb/smart mean in this context? 😄

File: src/IC-211/kotlin/io/kotest/plugin/intellij/files.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.search.FilenameIndex
+import com.intellij.psi.search.GlobalSearchScope
+import io.kotest.plugin.intellij.toolwindow.TagsFilename
+
+fun findFiles(project: Project): List<VirtualFile> {
+   return FilenameIndex
+      .getVirtualFilesByName(project, TagsFilename, false, GlobalSearchScope.allScope(project))
+      .toList()
+}
```
Comment (Kantis) [inline src/IC-211/kotlin/io/kotest/plugin/intellij/files.kt:13]: Identical on all platforms, can't it be put in `main` instead?

File: src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiTreeAnyChangeAbstractAdapter
+import io.kotest.plugin.intellij.findFiles
+import io.kotest.plugin.intellij.psi.getAllSuperClasses
+import org.jetbrains.kotlin.idea.core.util.toPsiFile
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtProperty
+
+const val TagsFilename = "KotestTags.kt"
```
Comment (Kantis) [inline src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt:14]: I suppose scanning all files would be too heavy on performance?

**Review by Kantis — APPROVED** [review]
Looks cool! Should I look into anything for the Gradle plugin?

File: src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt (hunk 1 above)
Comment (sksamuel) [reply src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt:49]: It's an intellij thing - it means does your code need the index or not.

File: src/IC-211/kotlin/io/kotest/plugin/intellij/files.kt (hunk 2 above)
Comment (sksamuel) [reply src/IC-211/kotlin/io/kotest/plugin/intellij/files.kt:13]: Good call.

File: src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt (hunk 3 above)
Comment (sksamuel) [reply src/main/kotlin/io/kotest/plugin/intellij/toolwindow/tags.kt:14]: Yes, and also we could run into "matches" which weren't really tags.
We can always start here and expand it.

**Pull request merged**